src/test/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctionsTest.scala

```diff
@@ -185,4 +191,199 @@ class PipelineFunctionsTest extends AnyFunSpec with DataFrameComparer with Spark
+        .select("converted_column")
+        .collect().head.get(0).toString.split("\\.").takeRight(1).head
+      )
+    }
+    it("should return null for unexpected timestamp format") {
+      Given("a dataframe with timestamp column of unexpected format")
+      val df = spark.sql("select '10/14/2016 09:28 PM' as ts")
+
+      When("function is called on this column")
+
+      Then("return a null value for the converted column")
+      assertResult(null) (df
+        .withColumn("converted_column", PipelineFunctions.epochMilliToTs("ts"))
+        .select("converted_column")
+        .collect().head.get(0)
+      )
+    }
+  }
+
+  describe("Tests for tsToEpochMilli") {
+    it("should convert timestamp to epoch time and preserve milliseconds") {
+      Given("a dataframe with timestamp ")
+      val df = spark.sql("select '2021-11-12 02:12:23.8870' as ts")
+
+      When("function is called on this column")
+
+      Then("convert timestamp to epoch time and preserve milliseconds")
+      assertResult("DoubleType") (df
+        .withColumn("epoch_milliseconds", PipelineFunctions.tsToEpochMilli("ts"))
+        .dtypes.filter(_._1 == "epoch_milliseconds").head._2
+      )
+      assertResult("887") (df
+        .withColumn("epoch_milliseconds", PipelineFunctions.tsToEpochMilli("ts").cast("Long").cast("String"))
+        .select("epoch_milliseconds").collect().head.get(0).toString.takeRight(3)
+      )
+    }
+    it("should return null for unexpected values") {
+      Given("a dataframe with unexpected timestamp value")
+      val df = spark.sql("select '1636663343887' as ts")
+
+      When("function is called on this column")
+
+      Then("return a null value for the converted column")
+      assertResult(null) (df
+        .withColumn("epoch_milliseconds", PipelineFunctions.tsToEpochMilli("ts"))
+        .select("epoch_milliseconds")
+        .collect().head.get(0)
+      )
+
+    }
+  }
+
+  describe("Tests for getDeltaHistory") {
+    it("should get the history of delta table") {
+      Given("a delta table")
+      val df = spark.range(10).withColumn("c1", lit("c1"))
+      df.write.format("delta").mode("overwrite").save("/tmp/overwatch/tests/table1")
+
+      val dummyConfig = new Config()
+      dummyConfig.setDatabaseNameAndLoc("dummy", "dummy", "/tmp/overwatch/tests")
+
+      val testTable: PipelineTable = PipelineTable(name = "table1", _keys = Array("id"), config = dummyConfig)
+
+      When("function is called on the given delta table")
+      val functionOutputDf: DataFrame = PipelineFunctions.getDeltaHistory(spark, testTable)
+
+      Then("returns a dataframe with detal table history")
+      val expectedDf: DataFrame = DeltaTable.forPath(testTable.tableLocation).history(9999)
+        .select("version", "timestamp", "operation", "clusterId", "operationMetrics", "userMetadata")
+
+      assertApproximateDataFrameEquality(functionOutputDf, expectedDf, 0.001)
```

Comment:
should we use `assertSmallDatasetEquality` instead of approximate? This is a tiny testDF I think approximate may not be appropriate here.